service/src/main/java/bio/terra/landingzone/library/landingzones/management/ResourcesReader.java

```diff
@@ -11,5 +11,9 @@ public interface ResourcesReader {
 
```

Comment:
I found myself a little confused by the method names here. Would something like this be clearer?

```
public interface ResourcesReader {
  default List<DeployedResource> listSharedResources() {
    return listResourcesByPurpose(ResourcePurpose.SHARED_RESOURCE);
  }

  List<DeployedResource> listResourcesByPurpose(ResourcePurpose purpose);

  List<DeployedResource> listResources();

  List<DeployedVNet> listVNetsByPurpose(SubnetResourcePurpose purpose);

  List<DeployedVNet> listVNets();
}
```

Reply:
I think this is good idea.
Initially, I wanted to emphasize that the list is not of ALL resources, but rather all that have known (defined in this library) tag. On a second thought, the implicit limitation is by the library design and doesn't have to be spelled out here.

Reply:
I agree that we need to revisit the names here. One key point is that the purpose is set to the subnet and not the VNet. 

Let me step back a bit, and walk through a simpler approach, that I am hoping could be worth considering. The initial interface was this:

```
public interface ResourcesReader {
  List<DeployedResource> listSharedResources();

  List<DeployedResource> listResourcesByPurpose(ResourcePurpose purpose);

  List<DeployedVNet> listVNetWithSubnetPurpose(SubnetResourcePurpose purpose);
}
```

A simple option to meet the requirement is for the service to use the methods `listResourcesByPurpose` and `listVNetWithSubnetPurpose` to create a unified list by calling these methods for each available purpose. The service should return a list of a new type (e.g. LandingZoneResource) if we are not doing this already. 

However, one challenge is that `listVNetWithSubnetPurpose` returns the VNet and all its subnets if one of them matches the requested purpose. This will make creating the list a bit cumbersome, albeit doable. So to simplify the creation of the unified list, we could add a new method to the reader, the interface would be:

```
public interface ResourcesReader {
  List<DeployedResource> listSharedResources();

  List<DeployedResource> listResourcesByPurpose(ResourcePurpose purpose);

  List<DeployedVNet> listVNetWithSubnetPurpose(SubnetResourcePurpose purpose);

 List<DeployedSubnet> listSubnetsByPurpose(SubnetResourcePurpose purpose);
}
```
Where `DeployedSubnet` contains the VNetID, Subnet name/id  and its purpose all in a flattened structure.